MentorMatchmaker/MatchMaking.fs

```diff
@@ -0,0 +1,134 @@
+﻿module MentorMatchmaker.MatchMaking
+
+open MentorMatchmaker
+open Flips
+open Flips.Types
+open Flips.SliceMap
+
+let private toTopicWeight topic : float =
+    let Anything = 1
+    let Common = 3
+    let Popular = 5
+    let Rare = 10
+
+    match topic with
+    | IntroductionToFSharp -> Common
+    | DeepDiveInFSharp -> Popular
+    | ContributeToOpenSource -> Popular
+    | WebDevelopment -> Popular
+    | ContributeToCompiler -> Rare
+    | MachineLearning -> Rare
+    | DistributedSystems -> Rare
+    | MobileDevelopment -> Rare
+    | DomainModelling -> Popular
+    | UpForAnything -> Anything
+    | DesigningWithTypes -> Rare
+    | MetaProgramming -> Rare
+
+type private ApplicantData = {
+    Id: int
+    Applicant: Applicant
+    Topics: Set<Topic>
+    WeekTimes: Set<WeekTime>
+}
+
+type private MentorId = int
+type private MenteeId = int
+
+let private toApplicantData id applicant =
+    { Id = id
+      Applicant = applicant
+      WeekTimes = toWeekTimes applicant.Availabilities
+      Topics = applicant.Topics }
+
+type UnpairedApplicants = ApplicantPool
+
```

Comment:
Would it be possible to have a summary explanation of how linear programming is used to make the connection between a mentor and a mentee?

Reply:
https://github.com/Kavignon/FSharp-Mentorship-Automatic-Planner/pull/26/commits/3fd50cd5df13f577a95fe4add0f9f45976842995